addons/hr_recruitment/models/hr_candidate.py

```diff
@@ -116,7 +116,7 @@ def _inverse_partner_email(self):
             if not candidate.partner_id:
                 if not candidate.partner_name:
                     raise UserError(_('You must define a Contact Name for this candidate.'))
-                candidate.partner_id = self.env['res.partner'].with_context(default_lang=self.env.lang).find_or_create(candidate.email_from)
+                candidate.partner_id = self.env['res.partner'].with_context(default_lang=self.env.lang).find_or_create(f"{candidate.partner_name} <{candidate.email_from}>")
```

Comment:
I think we should normalise the name and email like this for proper formatting, https://github.com/odoo/odoo/blob/ac5ea3cbd075db650c8e538d59e24657a0f04bb9/addons/website_event_booth/controllers/event_booth.py#L141-L142